client/src/views/files/FileOperationMenu.vue

```diff
@@ -196,16 +221,67 @@ function getFileOperationComponent(item: OperationItem): Component | undefined {
   }
 }
 
-function updateImportState(id: string, state: FileOperationState, stateData?: StateData): void {
-  const index = inProgressItems.value.findIndex((op) => op.data.id === id);
-  const operation = inProgressItems.value[index];
+function updateImportState(data: FileOperationData, state: FileOperationState, stateData?: StateData): void {
+  let index: number;
+
+  if (state === FileOperationState.OperationProgress) {
+    // Check if item has to be moved from queued to inProgress
+    if (!inProgressItems.value.find((op) => op.data.id === data.id)) {
+      switch (data.getDataType()) {
+        case FileOperationDataType.Import:
+          index = queuedUploadItems.value.findIndex((op) => op.data.id === data.id);
+          if (queuedUploadItems.value[index]) {
+            inProgressItems.value.push(queuedUploadItems.value[index]);
+            queuedUploadItems.value.splice(index, 1);
+          }
+          break;
+        case FileOperationDataType.Copy:
+          index = queuedCopyItems.value.findIndex((op) => op.data.id === data.id);
+          if (queuedCopyItems.value[index]) {
+            inProgressItems.value.push(queuedCopyItems.value[index]);
+            queuedCopyItems.value.splice(index, 1);
+          }
+          break;
+        case FileOperationDataType.Move:
+          index = queuedMoveItems.value.findIndex((op) => op.data.id === data.id);
+          if (queuedMoveItems.value[index]) {
+            inProgressItems.value.push(queuedMoveItems.value[index]);
+            queuedMoveItems.value.splice(index, 1);
+          }
+          break;
+      }
+    }
+  }
+
+  let currentOperationArray: Array<OperationItem> = [];
```

Comment:
You could init `currentOperationArray` to `inProgressItems.value` directly and store the index. To me, it's a bit easier to follow.

```typescript
let currentOperationArray = inProgressItems.value;
index = inProgressItems.value.findIndex((op) => op.data.id === data.id);

if (index === -1) {
  switch (data.getDataType()) {
    ...
  }
  index = currentOperationArray.findIndex((op) => op.data.id === data.id);
}

const operator = currentOperationArray[index];
```